pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -135,6 +153,50 @@ func (c ConfigObserver) sync(ctx context.Context, syncCtx factory.SyncContext) e
 	return configError
 }
 
+func (c ConfigObserver) updateObservedConfig(syncCtx factory.SyncContext, existingConfig map[string]interface{}, mergedObservedConfig map[string]interface{}) error {
+	existingConfigMaybeNested := map[string]interface{}{}
+	mergedObservedConfigMaybeNested := map[string]interface{}{}
+	if len(c.nestedConfigPath) > 0 {
```

Comment:
extract the filtering to a single method for me?  I think I want to re-use the filtering to avoid the defer handling we have in other spots.

Reply:
```go
filterConfig(nestedConfigPath, config, existingErr) (mergedObservedConfig map[string]interface{}, error){
}
```

called twice.

Reply:
```go

func configobserver() (map[string]string, error) {
	return map[string]string{"foo": "bar"}, nil
}

type filter struct {
	path []string
}

func (f filter) Filter(in map[string]string, err error) (map[string]string, error) {
	// do the filter
	return in, err
}

func main() {
	ret, err := filter{path: []string{"foo"}}.Filter(configobserver())

	fmt.Printf("%v and %v\n", ret, err)
}

```

this looks pretty good.  clean filter

Reply:
I see I wrote a non-sequitor.  I'll do that some other time.